roles/grafana/tasks/main.yml

```diff
@@ -19,14 +19,27 @@
 - name: Execute OS-specific Grafana installation
   ansible.builtin.include_tasks: "{{ ansible_facts['distribution'] }}.yml"
 
+- name: Check if Grafana admin password has already been set
+  ansible.builtin.stat:
+    path: /etc/grafana/.admin_password_set
+  register: grafana_admin_password_marker
+
```

Comment:
I suggest testing access to the HTTP/REST API first and on failure, running the CLI reset.